Creata una bozza della struttura di base.

NicFontana opened the pull request:
Sistemato il README
Aggiornato il gitignore

NicFontana committed:
Creata una bozza della struttura di base.
Sistemato il README
Aggiornato il gitignore

NicFontana commented:
Abbiamo pensato di abbandonare il DB, in favore di una classe di servizio PeerState che ci incapsula lo stato dell'applicazione. Come potete vedere, ci sono tutte le strutture dati pronte ad accogliere ciò che ci serve negli handler (che sono diventati due).

Commentate pls 😘

GiovanniFiorini commented:
ovviamente approvo 🐙

marcorambaldi commented on R00tella.py:
```diff
@@ -18,6 +15,13 @@ def server():
+	if not os.path.exists('downloads'):
+		os.mkdir('downloads')
+
+	p = Process(target=lambda: Server(3000, NeighboursHandler()).run())
```
Cosa sarebbe "lambda"?

NicFontana replied:
E' una keyword che ti permette di definire al volo una funzione. E' utile perché se devi passare una funzione come parametro ad un altra funzione, non c'è bisogno di fare
```python
def nome_funzione()
    #operazioni
```
e poi passare nome_funzione come parametro

marcorambaldi replied:
ah ok ho capito!

marcorambaldi reviewed (commented):
Direi che può funzionare!!

NicFontana committed:
Reso runnabile il progetto.
Separata la logica del menù da quella del peer: ora basta cambiare Menu per ottenere la grafica di un menù diverso, e MenuHandler per cambiare la logica che gestisce il menù.

FIAV1 commented on handler/MenuHandler.py:
```diff
@@ -0,0 +1,61 @@
+
+	def __kill_server(self, p: Process):
+		p.terminate()
+
```
```python
def broadcast(self, request:str)
    # recupera i vicini nel dizionario
    # crea socket attive per tutti i vicini
    # scrive sulle socket la request e chiude le socket attive
```

FIAV1 commented on handler/MenuHandler.py:
```diff
@@ -0,0 +1,61 @@
+from multiprocessing import Process
+
+
+class MenuHandler(HandlerInterface):
```
Non implementa più l'handler interface

FIAV1 commented on handler/MenuHandler.py:
```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+from handler.SelfHandler import SelfHandler
+from handler.HandlerInterface import HandlerInterface
```
Rimuovere l'import dell'interfaccia di handler

NicFontana committed:
Correzioni alla struttura
